File: backend/FlightBoard.Domain/Entites/Flight.cs
using FlightBoard.Domain.Enums;

namespace FlightBoard.Domain.Entities;

public class Flight
{
    public Guid Id { get; set; }
    public string FlightNumber { get; set; } = string.Empty;
    public string Destination { get; set; } = string.Empty;
    public DateTime DepartureTime { get; set; }
    public string Gate { get; set; } = string.Empty;
    public FlightStatus Status { get; set; }
    public string StatusString => Status.ToString();

    public void CalculateFlightStatus(DateTime now)
    {
        var diff = DepartureTime - now;

        if (diff.TotalMinutes <= 0 && diff.TotalMinutes > -60)
            Status = FlightStatus.Departed;
        else if (diff.TotalMinutes < 30 && diff.TotalMinutes >= 0)
            Status = FlightStatus.Boarding;
        else if (diff.TotalMinutes <= -60)
            Status = FlightStatus.Landed;
        else
            Status = FlightStatus.Scheduled;
    }
}

File: frontend/src/types/Flight.ts
export type FlightStatus = 'Scheduled' | 'Boarding' | 'Departed' | 'Landed';

export type FlightFormField =
  | 'flightNumber'
  | 'destination'
  | 'gate'
  | 'departureTime';

export interface Flight {
  id: string;
  flightNumber: string;
  destination: string;
  departureTime: string;
  gate: string;
  statusString: FlightStatus;
}

export interface FlightFormValidation {
  flightNumber?: string;
  destination?: string;
  gate?: string;
  departureTime?: string;
}

export interface FlightFormState {
  flightNumber: string;
  destination: string;
  gate: string;
  departureTime: string;
  isSubmitted?: boolean;
  errors: FlightFormValidation;
}

export interface FlightSearchState {
  destination: string;
  status: FlightStatus | '';
  searchResults: Flight[];
  searchIsActive: boolean;
}

export interface CreateFlightDto {
  flightNumber: string;
  destination: string;
  gate: string;
  departureTime: string;
}

export interface FlightEventsState {
  created: string[];
  updated: string[];
  deleted: string[];
}

export interface flightDetailsProps {
  flight: Flight;
  isNew: boolean;
  isUpdated: boolean;
}
